src/SerializerTrait.php

```diff
@@ -27,7 +27,8 @@ public function normalize($data, $format = null, array $context = [])
         $normalizedData = parent::normalize($data, $format, $context);
 
         if (\is_object($data)) {
-            $typeData = [self::KEY_TYPE => \get_class($data)];
+            $typeName = TypeMapper::getTypeByClass(\get_class($data));
```

Comment:
Same here, couldn't we inject the type mapper as a service instead of calling a static method?